tenets/codelingo/code-review-comments/structs-with-mutex/codelingo.yaml

```diff
@@ -0,0 +1,33 @@
+tenets:
+  - name: structs-with-mutex
+    flows:
+      codelingo/review:
+        comment: If the receiver is a struct that contains a sync.Mutex or similar synchronizing field, the receiver must be a pointer to avoid copying. From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#receiver-type)
+      codelingo/docs:
+        title: Structs with Mutex
+        body: If the receiver is a struct that contains a sync.Mutex or similar synchronizing field, the receiver must be a pointer to avoid copying.
+    query: |
+      import codelingo/ast/go
+
+      
+      go.type_spec(depth = any):
```

Comment:
We should scope this to a single package as "You cannot define a method on a type from another package". [cited from some random blog thing](http://www.golangbootcamp.com/book/methods).

Reply:
@BlakeMScurr How would we do that? Like this?

```
go.file(depth = any):
  go.ident:
    name as packageName
  ...
go.file(depth = any):
  go.ident:
    name == packageName
```